src/reim/reimFFTShift.js

```diff
@@ -0,0 +1,12 @@
+export function reimFFTShift(data) {
```

Comment:
This is not specific to FFT seems to me and look vary similar to a Rotate.

https://github.com/mljs/spectra-processing/blob/master/src/x/xRotate.js

Wonder if this should not be a rotate with a shift of length/2

I think also that xRotate should be improved and use the set as you do. It may be faster (not sure)

Reply:
yes, it could be used. but it can be used with typed arrays because it uses concat method and also it is 10 times slower

`
for 512 K points:

fftshift: 12.021240234375ms
xRotate: 122.616943359375ms`

Reply:
we could improve xRotate if you want. It is a generalized method indeed

Reply:
If we use this. it is faster
`
let result = new Float64Array(array.length); 

result.set(array.slice(array.length - shift)); 

result.set(array.slice(0, array.length - shift), shift);
`

Reply:
Yes you should update xRotate to use your approach. It is ok in this package to return (and convert) to Float64Array.

You may then rename reimFFTShift.js to reimRotate.js that uses actually xRotate.